src/component/FetchItems.js

```diff
@@ -78,19 +32,103 @@ const FetchItems = ({ token, setToken, firestore }) => {
         <Navbar />
       </React.Fragment>
     );
-  } else {
-    return (
-      <React.Fragment>
-        <SortedLists
-          token={token}
-          handlePurchase={handlePurchase}
-          calculateIfPurchased={calculateIfPurchased}
-        />
-        <Navbar />
-        <DeleteToken token={token} setToken={setToken} />
-      </React.Fragment>
-    );
   }
+
+  return (
+    <React.Fragment>
+      <section className="listFrame">
+        <h1 className="listTitle">
+          <span
+            role="img"
+            className="shoppingCart"
+            aria-label="Illustration of a shopping cart"
+          >
+            &#128722;
+          </span>
+          My Shopping List
+        </h1>
+        {/* could also use &#128717; which is shopping bags */}
+
+        <FirestoreCollection
+          path={concatPath}
+          sort="numberOfDays:asc"
+          render={({ isLoading, data }) => {
+            if (isLoading) {
+              return <div>Still Loading...</div>;
+            } else {
+              // this object will organize the list items into lists
+              const lists = {
+                soon: {
+                  items: [],
+                  className: 'soonItems',
+                  label: 'Soon Items',
+                },
+                prettySoon: {
+                  items: [],
+                  className: 'prettySoonItems',
+                  label: 'Pretty Soon Items',
+                },
+                notSoon: {
+                  items: [],
+                  className: 'notSoonItems',
+                  label: 'Not Soon Items',
+                },
+                inactive: {
+                  items: [],
+                  className: 'inactiveItems',
+                  label: 'Inactive Items',
+                },
+              };
```

Comment:
**No need to change:**

This object will be re-declared every time the component is rendered, is that what you want? I'd think it would be better to define it outside the component (unless you really want to initialize it on every render)

Reply:
@alejandronanez Is that a case where you'd lift state to the parent component or would it be sufficient to create the data structure outside the `return` statement? It's been a while since I've written much React. I'd have declared this structure as part of the `state` in the constructor, but it seems we no longer have constructors! 😬😅

Reply:
Great question @segdeha!

We could do both:

1. Move the data structure to the state using `React.useState`
2. Move the data structure outside of the return - even outside of the component itself will be better.

If I'd have to choose, I'd go with solution `#2` since it will require no changes to the rest of the existing code. Plus, we're passing a new data structure to each one of the children (`ListContent`) and that's triggering a re-render.

If we'd go with `#1`, we'd have to write more code because if you use hooks, it's better to have a separate `state` for each one of the lists. Updating state in hooks works different than `this.setState` in class components, hooks don't _merge_ your state. So this would be something like

```js
const [soonList, setSoonList] = React.useState([]);
const [prettySoonList, setPrettySoonList = React.useState([]);
const [notSoonList, setNotSoonList] = React.useState([]);
const [inactiveList, setInactiveList = React.useState([]);
```

Ps. Sorry for the super duper late response, 😅 I completely missed your comment!